projects/ids-enterprise-ng/src/lib/modal-dialog/soho-modal-dialog.service.ts

```diff
@@ -43,29 +41,12 @@ export class SohoModalDialogService {
    * The dialog won't necessarily be open yet, see the dialog ref api for further methods.
    *
    * @param component - the type of the component to instantiate; must not be null.
-   * @param parent - the parent container; must not be null.
-   * @param options - the control options to use.
    *
    * @return the modal dialog reference.
    */
-  public modal<T>(component: ComponentType<T>, parent: ViewContainerRef, options?: SohoModalOptions): SohoModalDialogRef<T> {
+  public modal<T>(component: ComponentType<T>): SohoModalDialogRef<T> {
```

Comment:
I suggest leaving the parent argument in place and making it optional and marking it deprecated for a period of time as this is also a big breaking change.

Reply:
yes this was because this now uses ApplicationRef instead of ViewContainerRef so that the passed component won't need to be displayed temporarily some placeholder somewhere before being  moved inside the modal content.

Reply:
I agree with @fitzorama on this, we need to keep the existing APIs until we increment the major version.

Reply:
That could line up with Ng 9 version 7/0 NG?

Reply:
The reason we used a ViewContainerRef (placeholder) was so that "nested" dialogs would work. If we attach a dialog to the ApplicationRef and then that dialog opens another dialog does attaching that to the ApplicationRef work?